engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/TypeOfNode.java

```diff
@@ -116,6 +116,16 @@ Object doPolyglotNumber(
     }
   }
 
+  @Specialization
+  Object doholeInAtom(AtomWithAHoleNode.HoleInAtom holeInAtom) {
+    Builtins builtins = EnsoContext.get(this).getBuiltins();
+    return Warning.attach(
+        EnsoContext.get(this),
+        builtins.nothing(),
+        Text.create("Hole in an atom doesn't have a type before being filled"),
+        null);
```

Comment:
That's OK, but wouldn't it be better to return a dataflow error?

Right now
```
Meta.type_of Nothing == Nothing == Meta.type_of atom.hole
```
so it is not easy to distinguish the situation where a hole in atom was queried vs where a `Nothing` type was. Of course one can do so by checking the warnings, but IMO a 'slightly more severe' dataflow error would suit better here. IMO it is an error to query a type of a _for all intents and purposes **uninitialized**_ value.